control/grpc.py

```diff
@@ -982,10 +982,27 @@ def construct_key_name_for_keyring(subsysnqn: str, hostnqn: str, prefix: str = N
     def remove_key_from_keyring(self, key_type: str, subsysnqn: str, hostnqn: str) -> None:
         key_name = GatewayService.construct_key_name_for_keyring(subsysnqn, hostnqn, key_type)
         assert self.rpc_lock.locked(), "RPC is unlocked when calling keyring_file_remove_key()"
+        key_list = []
         try:
-            rpc_keyring.keyring_file_remove_key(self.spdk_rpc_client, key_name)
+            key_list = rpc_keyring.keyring_get_keys(self.spdk_rpc_client)
         except Exception:
-            pass
+            self.logger.exception("Can't list keyring keys")
+            key_list = []
+
+        key_exists = False
+        for one_key in key_list:
+            try:
+                if one_key["name"] == key_name:
+                    key_exists = True
+                    break
+            except Exception:
```

Comment:
I think maybe we need 2 excepts? one with specific exception you expect to be raised, and a general one with Exception just to make sure this function won't raise exception - which will also log the exception?